cmd/neofs-node/object.go

```diff
@@ -522,3 +537,40 @@ func (c *reputationClientConstructor) Get(info coreclient.NodeInfo) (coreclient.
 
 	return cl, nil
 }
+
+type engineWithNotifications struct {
+	e  *engine.StorageEngine
+	nw notificationWriter
+	ns netmap.State
+
+	defaultTopic string
+}
+
+func (e engineWithNotifications) Put(o *objectSDK.Object) error {
+	if err := engine.Put(e.e, o); err != nil {
+		return err
+	}
+
+	ni, err := o.NotificationInfo()
+	if err == nil {
```

Comment:
Log the error if non-nil?

Reply:
Debug logs, perhaps. Log records produced by individual objects tend to consume all free space, unfortunately.

Reply:
it could contain two types of errors:
1. notification is not set for the object (unfortunately, `sdk` does not have dedicated err for that, only `api`; will add), we do not need to log that
2. parsing errors that are not expected since it was received from the node that approved it